backend/backend/authentication.py

```
@@ -0,0 +1,172 @@
+                firebase_uid=uid,
+                email=user_details['email'],
+                first_name=user_details['first_name'],
+                last_name=user_details['last_name'],
+                role=user_details['role']
+            )
+            logger.info(f"Successfully created local user for uid {uid} with role {user_details['role']}.")
+            self._create_profile(user, user_details['role'])
+            return user
+        except Exception as e:
+            # Catch potential database errors during user creation
+            logger.error(f"Database error creating user for uid {uid}: {e}", exc_info=True)
```

Ruff (0.8.2)

Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)